library/core/src/option.rs

```diff
@@ -2036,6 +2036,64 @@ impl<'a, T> const From<&'a mut Option<T>> for Option<&'a mut T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<T> crate::marker::StructuralPartialEq for Option<T> {}
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<T: PartialEq> PartialEq for Option<T> {
+    #[inline]
+    default fn eq(&self, other: &Self) -> bool {
```

Comment:
IIRC we're not supposed to put `default fn`s in `impl`s for public traits, as this would allow people to specialize this for their own things too.

So I think this should switch to calling a private `SpecOptionPartialEq` trait with the specializaiton.